include/tscpp/util/TsSharedMutex.h

```diff
@@ -53,6 +54,30 @@
 
 namespace ts
 {
+class Strerror
+{
+public:
+  Strerror(int err_num)
+  {
+    _c_str = strerror_r(err_num, _buf, 256);
+    if (!_c_str) {
```

Comment:
some docs show that with the GNU-specific `strerror_r()`, if `err_num` is unknown, it can return an "unknown error nnn" message which won't make `_c_str` null, making the conditional think that `strerror_r()` did not fail when it did. Maybe use the XSI-compliant version (returning int) and returning `_buf` which also has the error string

Reply:
TS is compiled with -D_GNU_SOURCE, which means that only the GNU-specific strerror_r() is available.  Even if it never returns null (currently), I'd rather leave the handling in for future-safety.

Reply:
This breaks build on macOS https://github.com/apache/trafficserver/issues/8642